streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -227,6 +230,27 @@ public void initializeIfNeeded() {
         }
     }
 
+    private void initOffsetsIfNeeded(final java.util.function.Consumer<Set<TopicPartition>> offsetResetter) {
+        final Map<TopicPartition, OffsetAndMetadata> committed = mainConsumer.committed(resetOffsetsForPartitions);
+        for (final Map.Entry<TopicPartition, OffsetAndMetadata> committedEntry : committed.entrySet()) {
+            final OffsetAndMetadata offsetAndMetadata = committedEntry.getValue();
+            if (offsetAndMetadata != null) {
+                mainConsumer.seek(committedEntry.getKey(), offsetAndMetadata);
+                resetOffsetsForPartitions.remove(committedEntry.getKey());
+            }
+        }
+
+        if (!resetOffsetsForPartitions.isEmpty()) {
```

Comment:
this `if` is not strictly required, however, it allows us to just pass `null` as `offsetResetter` in tests, so might be worth it.

Reply:
Can we just pass in a no-op lambda instead? I'd rather avoid special handling for null input that isn't supposed to be null, just so we can use null in the tests (which are therefore not realistic tests since it should never be null, no?)

Reply:
Fair enough.